inventorization_app/views.py

```diff
@@ -1,120 +1,71 @@
-                                status=status.HTTP_202_ACCEPTED)
-            else:
-                return Response(serializer.errors,
-                                status=status.HTTP_400_BAD_REQUEST)
-        else:
-            serializer = Api_serializer(data=product)
-            if serializer.is_valid():
-                return Response(serializer.data, 
-                                status=status.HTTP_202_ACCEPTED)
-            else:
-                return Response(serializer.errors,
-                                status=status.HTTP_400_BAD_REQUEST)
-            
-class DenyOwnerView(viewsets.ViewSet):  
+from rest_framework.response import Response
 
-    def retrive(self, request, user, token, pk): 
-        queryset = Products.objects.filter(id=pk)
-        product = get_object_or_404(Products, id=pk)
-        user = get_object_or_404(User, id=user)
-    
-        serializer = Api_serializer(data=product)
-        if serializer.is_valid():
-            return Response(serializer.data, 
-                            status=status.HTTP_202_ACCEPTED)
-        else:
-            return Response(serializer.errors,
-                            status=status.HTTP_400_BAD_REQUEST)  
+from .models import Products
+from .permissions import IsOwnerOrReadOnly, IsAdmin
+from .permissions import permision_or
+from .serializers import AcceptSerializer
+from .serializers import ApiSerializer
+from .serializers import DenySerializer
+from .serializers import EmailSerializer
 
 
-class ProductListView(generics.ListAPIView):
+class ProductViewSet(viewsets.ModelViewSet):
     queryset = Products.objects.all()
-    serializer_class = Api_serializer
-    permission_classes = (IsAuthenticated,)
-    filter_backends = (filters.SearchFilter, django_filters.rest_framework.DjangoFilterBackend)
+    serializer_class = ApiSerializer
+    filter_backends = (filters.SearchFilter, DjangoFilterBackend)
     search_fields = ('product_key', 'name', 'category')
     filter_fields = ('product_key', 'name', 'category')
+    permission_classes = (IsAuthenticated, permision_or(IsAdmin, IsOwnerOrReadOnly))
 
+    def validate_and_response(self, serializer, data):
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
-class ProductCreateView(generics.CreateAPIView):
-    queryset = Products.objects.all()
-    serializer_class = Api_serializer
-    permission_classes = (IsAdminUser,)
-
+    @action(methods=['post'], detail=True, url_path=r'accept/(?P<user_id>\d+)/(?P<token>\w+)', url_name='accept',
+            permission_classes=(IsAdminUser,))
+    def accept_owner(self, request, user_id, pk, token):
+        data = {'user_id': user_id,
+                'product_id': pk,
+                'token': token, }
+        serializer = AcceptSerializer(data=data)
```

Comment:
http://www.cdrf.co/3.7/rest_framework.viewsets/ModelViewSet.html#get_serializer_class

self.action ? (action in view)

make if which return good serializer for each viewset action

Reply:
serializer = self.get_serilizer(data=data)
serialzier.is_valid(?)
return Response(serializer.data, status=status.HTTP_200_OK)